queries/php/highlights.scm

```diff
@@ -0,0 +1,191 @@
+; Types
+
+[
+ (primitive_type)
+ (cast_type)
+ ] @type.builtin
+(type_name (name) @type)
+
+; Functions
+
+(array_creation_expression "array" @function.builtin)
+(list_literal "list" @function.builtin)
+
+(method_declaration
+  name: (name) @method)
+
+(function_call_expression
+  function: (qualified_name (name)) @function)
+
+(scoped_call_expression
+  name: (name) @function)
+
+(member_call_expression
+  name: (name) @method)
+
+(function_definition
+  name: (name) @function)
+
+; Member
+
+(property_element
+  (variable_name) @property)
+
+(member_access_expression
+  name: (variable_name (name)) @property)
+(member_access_expression
+  name: (name) @property)
+
+; Variables
+
+(relative_scope) @variable.builtin
+
+((name) @constant
+ (#match? @constant "^_?[A-Z][A-Z\d_]+$"))
+
+((name) @constructor
+ (#match? @constructor "^[A-Z]"))
+
+((name) @variable.builtin
+ (#eq? @variable.builtin "this"))
+
+(variable_name) @variable
+
+; Basic tokens
+
+[
+ (string)
+ (heredoc)
+ ] @string
+
+(boolean) @boolean
+(null) @constant.builtin
+(integer) @number
+(float) @float
+(comment) @comment
+
+"$" @constant
+
+; Keywords
+
+[
+ "abstract"
+ "as"
+ "break"
+ "class"
+ "const"
+ "continue"
+ "declare"
+ "default"
+ "echo"
+ "enddeclare"
+ "extends"
+ "final"
+ "function"
+ "global"
+ "implements"
+ "insteadof"
+ "interface"
+ "namespace"
+ "new"
+ "private"
+ "protected"
+ "public"
+ "return"
+ "static"
+ "trait"
+ ] @keyword
+
+[
+ "case"
+ "else"
+ "elseif"
+ "endif"
+ "endswitch"
+ "if"
+ "switch"
+ ] @conditional
+
+[
+ "do"
+ "endfor"
+ "endforeach"
+ "endwhile"
+ "for"
+ "foreach"
+ "while"
+ ] @repeat
+
+[
+ "catch"
+ "finally"
+ "throw"
+ "try"
+ ] @exception
+
+[
+ "include_once"
+ "include"
+ "require_once"
+ "require"
+ "use"
+ ] @include
+
+[
+ ","
+ ";"
+ ] @punctuation.delimiter
+
+[
+ (php_tag)
+ "?>"
```

Comment:
Those two look weird to me, I guess `php_tag` is the whole node, and you might want to capture only the start and end of the tag using `(php_tag ... @punctuation.bracket)`.

Reply:
Thank you for your advice!
I tried your advice, but the tree-sitter got the following error:
```
Error detected while proccessing /path/to/nvim-treesitter.vim
line 23:
E5108: Error executing lua .../runtime/lua/vim/treesitter.lua:151: query: invalid syntax as position 1700
```

And when I checked the sample php program and its tree-sitter structure, it became as follows.

Sample program:
```php
<?php
$title = "sample";
?>

<html>
    <head>
        <title><?= $title ?></title>
    </head>
</html>
```

Tree sitter structure:
```
program [0, 0] - [0, 9])
  php_tag [0, 0] - [5, 0])
  expression_statement [0, 1] - [18, 1])
    assignment_expression [0, 1] - [17, 1])
      variable_name [0, 1] - [6, 1])
        name [1, 1] - [6, 1])
      string [9, 1] - [17, 1])
  text_interpolation [0, 2] - [18, 6])
    text [0, 4] - [15, 6])
    php_tag [15, 6] - [18, 6])
  expression_statement [19, 6] - [25, 6])
    variable_name [19, 6] - [25, 6])
      name [20, 6] - [25, 6])
  text_interpolation [26, 6] - [0, 9])
    text [28, 6] - [0, 9])
```

I think that "php_tag" is not whole node, isn't it?
I could be wrong, and if so, please point me out.

Reply:
I just tested it "php_tag" is just the start, not the whole node.